protocol/src/main/scala/io/mediachain/protocol/CborSerialization.scala

```diff
@@ -693,8 +693,8 @@ object CborSerialization {
 
     override def fromCValue(cValue: CValue): Xor[DeserializationError, MultihashReference] = {
       cValue match {
-        case CTaggedValue(tag, CBytes(value)) if tag == MultihashReference.cborLinkTag =>
-          MultiHash.fromBytes(value)
+        case CTaggedValue(tag, CString(value)) if tag == MultihashReference.cborLinkTag =>
+          MultiHash.fromBase58(value)
```

Comment:
hmm, just read through [the spec](https://github.com/ipfs/specs/tree/master/ipld#serialised-cbor-with-tags), and it seems the js behavior is correct:

- If the link is a valid multiaddress and converting that link text to the multiaddress binary string and back to text is guaranteed to result to the exact same text, the link is converted to a binary multiaddress stored in CBOR as a byte string (major type 2).
- Else, the link is stored as text (major type 3)

So we might need to use multiaddr after all if we want efficient storage.  The multiaddr header only adds about 3 bytes.  I've got the scala multiaddr port mostly working, but i need to do an error-handling pass and run some tests.